script/eslint.ts

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env ts-node
+
+import ChildProcess = require('child_process')
+import Path = require('path')
+
+const ESLINT_ARGS = [
+  '--cache',
+  '--rulesdir=./eslint-rules',
+  './{script,eslint-rules}/**/*.{j,t}s?(x)',
+  './tslint-rules/**/*.ts',
+  './app/{src,typings,test}/**/*.{j,t}s?(x)',
+  ...process.argv.slice(2),
+]
+const opts = {
+  stdio: 'inherit',
+}
+
+if (process.env.CI) {
+  ChildProcess.spawn(
+    Path.resolve(__dirname, '..', 'node_modules', '.bin', 'eslint'),
```

Comment:
Is there a reason why this is preferred over [the CLI interface](https://eslint.org/docs/4.0.0/developer-guide/nodejs-api)? 

I'm not sure what `process.argv.slice(2)` is used for here but this sort of example using [`CLIEngine`](https://eslint.org/docs/4.0.0/developer-guide/nodejs-api#cliengine) seems like it might be portable:

```js
var CLIEngine = require("eslint").CLIEngine;

var cli = new CLIEngine({
    cache: true,
    rulepath='./eslint-rules'
});

var report = cli.executeOnFiles(paths);
```